src/app/detail/[task_id]/[user_type]/page.tsx

```diff
@@ -102,12 +124,14 @@ export default async function TaskDetailPage({
   const { task_id } = params
   const copilotClient = new CopilotAPI(token)
 
-  const [task, assignee, attachments, activities, tokenPayload] = await Promise.all([
+  const [workflowStates, task, assignee, attachments, activities, tokenPayload, tasks] = await Promise.all([
+    getAllWorkflowStates(token),
     getOneTask(token, task_id),
     addTypeToAssignee(await getAssigneeList(token, params.user_type)),
     getAttachments(token, task_id),
     getActivities(token, task_id),
     copilotClient.getTokenPayload(),
+    getAllTasks(token),
```

Comment:
how come we do getAllTasks on details page?

Reply:
We are doing `getAllTasks` here for the real time functionality of the tasks details page. When a task title is updated, supabase event is listened in the `RealTime` hoc where we also update the `tasks` store with the updated task. After that, we update the `tasks` state in the redux store which is filtered out with the `task_id` in the details page to show the updated `title` in real time. In contrast, we need `tasks` state which we update from `getAllTasks`. 
We have to fetch `getAllTasks` here because next js doesn't load the page of `/` route and doesn't fetch tasks and update the `tasks` state if users access the `/details` route.